### PR TITLE
Ab#8252 carousel button box super wide buttons

### PR DESCRIPTION
**Pull request by PeculiarGoat**
ADO card: ☑️ [AB#8252](https://dev.azure.com/S72/SHIFT72/_workitems/edit/8252)

Designs: 
- 📱 [Mobile](https://app.zeplin.io/project/607f961cf9f03790cc66e475/screen/6205cb36667eccb5f9ae028f)
- 💊 [Tablet](https://app.zeplin.io/project/607f961cf9f03790cc66e475/screen/6205cb31c677b1be9ce9955c)
- 🖥 [Desktop](https://app.zeplin.io/project/607f961cf9f03790cc66e475/screen/6205cb301e8ac288b07757ee)

## Description of work
There was an issue where some buttons on the carousel could grow to be too wide under certain circumstances.  This PR solves that issue by refactoring and simplifying the button container.  Now, there is a generic `cta_buttons.jet` and `_cta-buttons.scss` used by the carousel and media detail pages which means we only need to edit HTML/CSS in one place whenever we change or add anything.  It also means that changes/additions in the cta button container will be less prone to creating unexpected side-effects/bugs.

## Edge cases/Caveats/Known issues
- In `_cta-buttons.scss`, there is a `#region` which overrides styles for the trailer, wishlist, and share buttons to better reflect the designs.  Once the new icon work has been completed for these buttons, that `#region` should be removed as it makes more sense for button styles to be defined in `_buttons.scss`.
- Relish issue [#305](https://github.com/indiereign/shift72-web/issues/305) can create a state where the `s72-pricing-buttons` component renders wrappers with no buttons inside.  This creates a `gap` indent before the trailer/wishlist/share button container.  This will be solved by a Relish fix with no change required in core-template.

## Checklist
- [x] CI tests are passing Github actions (inc. linting)
- [x] Key areas of the feature outlined for context and testing
- [x] If there are designs for this work are they noted here and in the ADO card 
- [x] Have checked this at multiple screen resolutions and range of browsers
- [x] Moved ADO card to Dev/done, checked link to Github, tagged "Review"
- [x] Updated changelog (if applicable)
- [x] I promise to document any new feature toggles/configurations in the appropriate documentation

### Jets
- Added `/common/cta_buttons.jet`.
  - Contains pricing buttons, play button, live event button, trailer button, wishlist button, and share modal button.
  - Simplifies some of the HTML and some of the CSS classes.
  - Each component/section is conditionally rendered.  This makes the rendered DOM smaller e.g. a carousel item containing a content page has no use for any of these buttons so they don't get rendered.  It will also discourage developers from making things harder to maintain by adding structural CSS to Relish component wrappers.
- Changed `/bundle/item.jet`, `/collection/carousel/item.jet`, `/film/item.jet`, and `/tv/detail.jet`.
  - Button HTML has been replaced by `cta_buttons.jet` for consistency.
  - Imports tidied up.
- Deleted `/bundle/buttons.jet`, `/bundle/extras.jet`, `/collection/carousel/item/buttons.jet`, `/collection/carousel/item/extras.jet`.
  - These files are made redundant by `cta_buttons.jet`.
- Deleted `/collection/carousel_item.jet`.
  - This file was deleted as part of the new carousel work but somehow has found it's way back into git.  It isn't used anymore.
- Deleted `/items/featured_item.jet`.
  - This file isn't used for anything.
  - Also removed the import from `/page/homepage.jet`.  It was imported but not used.
- Misc change to `/collection/carousel/item.jet`.
  - Added a check for `"Tagline"` before rendering the synopsis element.  I noticed that site builds would fail without this check if an episode was added to the carousel collection.

### CSS
- Added `_cta-buttons.scss`.
  - Contains structural CSS for new button box.
  - Contains style overrides for trailer, wishlist, and share buttons.  This should be removed when new icon work is done.
- Changed `_buttons.scss`.
  - Simplified trailer button CSS.
- Changed `_can-be-watched-button.scss`, `_carousel.scss`, `_meta-detail.scss`.
  - Removed old button box CSS.
- Changed `_plans.scss`.
  - Added margins to subscription trial label to match the designs.
- Changed `main.scss`.
  - Added import for `_cta-buttons.scss`.

### JS
- Changed `main.js`.
  - Changed the selector used to add event listeners to carousel trailer buttons as the CSS classes for that have been simplified.  I think it may be worth raising an issue in Relish to change the trailer button component so it can handle clicks from the carousel (to redirect and autoplay).  Seems a bit silly having two different trailer buttons.
  - `main.js.map` is automatically changed due to this.

### CODE REVIEW

**Comment by PeculiarGoat** [comment]
@stajs @additiverich FYI.

Tagging @curtisadams28 @matthewthomsonnz @twln1 for review as they have recently touched on these files.

**Comment by twln1** [comment]
Carousel has some indentation on the extra buttons for mobile views
image

**Comment by PeculiarGoat** [comment]
@twln1 The extra buttons container has margins on the left/right as shown in the design: https://app.zeplin.io/project/607f961cf9f03790cc66e475/screen/6205cb36667eccb5f9ae028f

**Review by curtisadams28 — APPROVED** [review]
LGTM 👍

File: site/styles/_plans.scss
Comment (matthewthomsonnz) [inline site/styles/_plans.scss:97]: That 18px padding causes the label to be offset 
image

**Review by matthewthomsonnz — APPROVED** [review]
Noticed one thing but otherwise looks good

File: site/styles/_plans.scss
Comment (PeculiarGoat) [reply site/styles/_plans.scss:97]: That's how it is in [the design](https://app.zeplin.io/project/607f961cf9f03790cc66e475/screen/6205cb31c677b1be9ce9955c).  Is that wrong?

**Comment by PeculiarGoat** [comment]
I'm going to block this until Curtis' carousel-positioning branch is merged just incase there are conflicts.  I would also like some approval/rejection from @twln1 please!

**Comment by stajs** [comment]
@PeculiarGoat if @twln1 is willing to defer other reviewers, I think that is enough for approve/reject.

File: CHANGELOG.md
Comment (stajs) [inline CHANGELOG.md:10]: I think explain the CTA acroynm for externals.

File: site/styles/_plans.scss
Comment (stajs) [reply site/styles/_plans.scss:97]: If I was impersonating a designer, I'd suggest it's probably meant to be centered, but there isn't a good solution for centering with translations. I thought there was some discussion about a left-margin and the text flowing under the other buttons?

**Review by stajs — APPROVED** [review]
A nice tidy up!

File: site/styles/_plans.scss
Comment (PeculiarGoat) [reply site/styles/_plans.scss:97]: I didn't consider that they might just want it centered.  RE text flowing under the buttons... it works fine for long translations with this implementation!

Comment (stajs) [reply site/styles/_plans.scss:97]: Yeah that's what I meant 👍

Comment (matthewthomsonnz) [reply site/styles/_plans.scss:97]: > If I was impersonating a designer, I'd suggest it's probably meant to be centered, but there isn't a good solution for centering with translations. I thought there was some discussion about a left-margin and the text flowing under the other buttons?

Theo and I figured out a strategy to make it centered until its the width of the subscribe button. Then it grows to the right.

**Pull request merged**
